contracts/src/defuse/payload/multi.rs

```diff
@@ -36,8 +55,26 @@ where
 
     fn try_from(value: MultiStandardPayload) -> Result<Self, Self::Error> {
         match value {
-            MultiStandardPayload::Nep413(payload) => payload.try_into(),
-            MultiStandardPayload::Erc191(payload) => serde_json::from_str(&payload),
+            MultiStandardPayload::Nep413(payload) => payload.payload.try_into(),
+            MultiStandardPayload::Erc191(payload) => serde_json::from_str(&payload.payload),
+            MultiStandardPayload::RawEd25519(payload) => payload.try_into(),
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use near_sdk::bs58;
+
+    use super::*;
+
+    #[test]
+    fn test_json() {
+        let p: MultiStandardPayload = serde_json::from_str(r#"{"standard":"raw_ed25519","payload":"{\"signer_id\":\"74affa71ab030d400fdfa1bed033dfa6fd3ae34f92d17c046ebe368e80d53751\",\"verifying_contract\":\"intents.near\",\"deadline\":{\"timestamp\":1732035219},\"nonce\":\"XVoKfmScb3G+XqH9ke/fSlJ/3xO59sNhCxhpG821BH8=\",\"intents\":[{\"intent\":\"token_diff\",\"diff\":{\"nep141:base-0x833589fcd6edb6e08f4c7c32d4f71b54bda02913.omft.near\":\"-1000\",\"nep141:eth-0xdac17f958d2ee523a2206206994597c13d831ec7.omft.near\":\"998\"}}]}","public_key":"ed25519:8rVvtHWFr8hasdQGGD5WiQBTyr4iH2ruEPPVfj491RPN","signature":"ed25519:3vtbNQJHZfuV1s5DykzyjkbNLc583hnkrhTz57eDhd966iqzkor6Twgr4Loh2C195SCSEsiGfrd6KcxpjNq9ZbVj"}"#).unwrap();
+        assert_eq!(
+            bs58::encode(p.hash()).into_string(),
+            "8LKE47o44ybZQR9ozLyDnvMDTh4Ao5ipy2mJWsYByG5Q"
+        );
+        p.verify().unwrap();
```

Comment:
```suggestion
        matches!(p, MultiStandardPayload::RawEd25519(p) if p.verify() == Some(p.public_key));
```

Reply:
there is no `p.public_key`, but I've added a similar test in next commit

Reply:
My version works as well because there is the `matches!` which unpacks  the `MultiStandardPayload`.